libkbfs/keybase_service_base.go

```diff
@@ -425,7 +425,15 @@ func (k *KeybaseServiceBase) LoadUserPlusKeys(ctx context.Context,
 	uid keybase1.UID, pollForKID keybase1.KID) (UserInfo, error) {
 	cachedUserInfo := k.getCachedUserInfo(uid)
 	if cachedUserInfo.Name != libkb.NormalizedUsername("") {
-		return cachedUserInfo, nil
+		if pollForKID == keybase1.KID("") {
+			return cachedUserInfo, nil
+		}
+		// Skip the cache if pollForKID isn't present.
```

Comment:
This was sort of a pre-existing bug -- we should be busting our local cache if the key isn't present.  However, `KBPKIClient.HasVerifyingKey` already took care of it on that path.  This way is better though.